sklearn/metrics/pairwise.py

```diff
@@ -767,6 +769,72 @@ def rbf_kernel(X, Y=None, gamma=None):
     return K
 
 
+def matern_kernel(X, Y=None, gamma=None, coef0=1.5):
+    """ Compute the Matern kernel between X and Y.
+
+    The class of Matern kernels is a generalization of the RBF and
+    absolute exponential kernel parameterized by an additional parameter
+    coef0 (commonly denoted as nu in the literature). The smaller coef0,
+    the less smooth the approximated function is. For nu=inf, the kernel
+    becomes equivalent to the RBF kernel and for nu=0.5 to the absolute
+    exponential kernel. Important intermediate values are nu=1.5 (once
+    differentiable functions) and nu=2.5 (twice differentiable functions).
+
+    See Rasmussen and Williams 2006, pp84 for details regarding the
+    different variants of the Matern kernel.
+
+    Parameters
+    ----------
+    X : array of shape (n_samples_X, n_features)
+
+    Y : array of shape (n_samples_Y, n_features)
+
+    gamma : float
+
+    coef0 : float>0.0 (the parameter nu)
```

Comment:
All parameter descriptions in this file seem to ignore the standard. Do you think we should change that in this or another pull request?